src/wrapper/wrapper.c

```diff
@@ -90,6 +90,11 @@ int sync_directory (const char* path);
  *                                                                           *
  *****************************************************************************/
 
+static inline bool is_wronly (int fd)
+{
+  return (O_WRONLY == fcntl(fd, F_GETFL) & O_ACCMODE);
```

Comment:
@JaeseungYeom this function currently won't work as expected. In C/C++, bitwise and (&) has a lower operator precedence than equality (==). So, right now, this line will compare the output of `fcntl` with `O_WRONLY` and then perform a logical and with `O_ACCMODE`. You need to put `fcntl(fd, F_GETFL) & O_ACCMODE` in parentheses to get the expected behavior.

Reply:
If I may insert a drive by comment also:

Suggestion: check for fcntl failure (-1) before the bitwise stuff and return false or whatever.  Even if an error is unlikely/impossible/irrelevant in the contexts where this function is called, not checking for the error will set off red flags for some people.

Reply:
> If I may insert a drive by comment also:
> 
> Suggestion: check for fcntl failure (-1) before the bitwise stuff and return false or whatever. Even if an error is unlikely/impossible/irrelevant in the contexts where this function is called, not checking for the error will set off red flags for some people.

An excellent suggestion. Thank you!

Reply:
> @JaeseungYeom this function currently won't work as expected. In C/C++, bitwise and (&) has a lower operator precedence than equality (==). So, right now, this line will compare the output of `fcntl` with `O_WRONLY` and then perform a logical and with `O_ACCMODE`. You need to put `fcntl(fd, F_GETFL) & O_ACCMODE` in parentheses to get the expected behavior.

Can you fix it and address Jim's comment below?

Reply:
@JaeseungYeom just pushed the fix. The `is_wronly` function will now return an integer. It will return -1 on `fcntl` error, 0 if the file descriptor is not write-only, and 1 if the file descriptor is write-only. I've also updated the spots where `is_wronly` is used so that, if an error occurs, a message will be printed with `DPRINTF`, and the actual file I/O call will be invoked (as opposed to calling all the DYAD-specific stuff).

I'm working on testing right now.